src/components/accession2/CalculatorModal.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import strings from 'src/strings';
+import Button from '../common/button/Button';
+import DialogBox from '../common/DialogBox/DialogBox';
+import { Grid } from '@mui/material';
+import { SelectT, Textfield } from '@terraware/web-components';
+import { Accession2, updateAccession2 } from 'src/api/accessions2/accession';
+import { ServerOrganization } from 'src/types/Organization';
+import { Unit, WEIGHT_UNITS_V2 } from '../seeds/nursery/NewTest';
+import useSnackbar from 'src/utils/useSnackbar';
+
+export interface CalculatorDialogProps {
+  open: boolean;
+  record: Accession2;
+  setRecord: React.Dispatch<React.SetStateAction<Accession2>>;
+  onClose: () => void;
+  reload: () => void;
+  organization: ServerOrganization;
+  onChange: (id: string, value: unknown) => void;
+  openPreviousModal: () => void;
```

Comment:
call it `onPrevious`? this component shouldn't know whether previous ui was a modal or something else.